src/any_llm/providers/zai/utils.py

```
@@ -0,0 +1,38 @@
+# enumerates the values without defining them: model_context_window_exceeded arrives
+# on a 200 mid generation with content already delivered, which is truncation, and
+# sensitive is a safety interception, which is what content_filter reports.
+# network_error is deliberately absent: a transport failure is not a completion
+# reason and no legal value fits.
+_FINISH_REASON_MAP: dict[str, Literal["length", "content_filter"]] = {
+    "model_context_window_exceeded": "length",
+    "sensitive": "content_filter",
+}
+
+
+def _normalize_finish_reason(response: Any) -> None:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `response`

(ANN401)